### PR TITLE
[RUN-3009] Keep track of `ReplayScriptAlive` and error-out `executeCommand` calls if its not

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-3009/error-no-context-group-available-for-isolate#comment-709b22c2
* [Tests](https://buildkite.com/replay/chromium-build/builds/2912): ✅
* Spot-checked: ✅
  * I manually ran `npm test -- replay-api/basics/run-2965.spec.ts` (which opens a page and then navigates twice)
    * → It produces 3 x `STATUS_CHANGE_ALIVE` + 2 x `STATUS_CHANGE_UNALIVE`
  * Experimented on pages with **cross-domain navigation** manually ✅

### CODE REVIEW

**Comment by Domiii** [comment]
TODO: Fix cross-domain navigation.

File: third_party/blink/renderer/bindings/core/v8/local_window_proxy.cc [hunk 1]
```diff
@@ -233,24 +232,24 @@ void LocalWindowProxy::Initialize() {
       InitializeRecordReplay(GetIsolate(), GetFrame(), context);
     }
 
-    if (doInit || GetFrame() == gRecordReplayFrame) {
```
Comment (Domiii) [inline third_party/blink/renderer/bindings/core/v8/local_window_proxy.cc:236]: Note: This did not handle potential cross-domain (and maybe other types of?) navigation where the frame does not get re-used.

**Review by toshok — APPROVED** [review]
feels like this PR is circling similar issues to what I'm running into with #1042

**Comment by Domiii** [comment]
> feels like this PR is circling similar issues to what I'm running into with #1042

Its rather different. The issue you were trying to fix is making certain aspects of shutdown more deterministic. What we are doing here is track liveness of our root frame, and with it of our `gReplayScript` and debugger sessions. This is less a concern of determinism and more about knowing why certain commands fail at certain points in time.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 2]
```diff
@@ -5643,16 +5685,26 @@ void OnRootFrameInit(v8::Isolate* isolate, LocalFrame* localFrame, v8::Local<v8:
...
+  if (gReplayScriptsAlive) {
+    // Our "V8RecordReplaySetDefaultContext" logic implies a single local
+    // root frame per render process.
+    recordreplay::Warning("ReplayScript Multiple_OnRootFrameInit");
+    return;
+  }
   
-  // NOTE: The root `LocalFrame` will actually not change over time.
-  gLocalRootFrame = localFrame;
+  // NOTE: The root `LocalFrame` will not necessarily change over time.
```
[the remark this answers is not in the document]
Comment (Domiii) [reply third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:5647]: Its also less wordy!

**Pull request merged**